src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,99 @@
+package duke.util;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.System.exit;
+
+public class Storage {
+
+    /**
+     * Checks if it's possible to get the file stored at Documents/Duke/dukeData.txt
+     * Create the file if necessary and possible
+     * @param storedItems
+     */
+    public void getDataFile(ArrayList<Task> storedItems) {
+        String home = System.getProperty("user.home");
+
+        Path dirPath = Paths.get(home, "Documents", "Duke");
+        Path filePath = Paths.get(home, "Documents", "Duke", "dukeData.txt");
+
+        if (Files.exists(dirPath)) {
+            if (Files.notExists(filePath))
+                try {
+                    Files.createFile(filePath);
+                } catch (IOException e) {
+                    System.out.println(e);
+                    exit(1);
+                }
+
+            readDataFile(filePath, storedItems);
+        } else {
+            try {
+                Files.createDirectories(dirPath);
+            } catch (IOException e) {
+                System.out.println(e);
+                exit(1);
```

Comment:
I'm not sure if we should force exit the code as such.. May need to ask the TA for clarification on this point.